Shared/Samples/Display points using clustering feature reduction/DisplayPointsUsingClusteringFeatureReductionView.swift

```diff
@@ -60,28 +60,30 @@ struct DisplayPointsUsingClusteringFeatureReductionView: View {
                 .task(id: identifyScreenPoint) {
                     guard let identifyScreenPoint,
                           let layer,
-                          let identifyResult = try? await proxy.identify(
-                            on: layer,
-                            screenPoint: identifyScreenPoint,
-                            tolerance: 3
-                          ),
-                          let firstPopup = identifyResult.popups.first
-                    else { return }
-                    self.popup = firstPopup
-                    self.showsPopup = true
+                          let identifyResult = await Result(awaiting: {
+                              try await proxy.identify(
+                                on: layer,
+                                screenPoint: identifyScreenPoint,
+                                tolerance: 3
+                              )
+                          })
+                        .cancellationToNil()
```

Comment:
Why wrap the result in another result?

Reply:
It looks like it is for when the result is a failure and the error is a cancellation error then the result should be `nil`. But I am also not sure what the use case is for that.

Reply:
I added these Result wrappers because the FloatingPanel changes were working fine in the Toolkit's Popup example but not in these samples. `popup` was still `nil` at the time of the force unwrap. 

The missing Result wrappers was the main difference I spotted between the two implementations.

Reply:
That's kind of strange. I don't expect the popup state property to be changed until the first identify operation runs. 🤔

Reply:
> That's kind of strange. I don't expect the popup state property to be changed until the first identify operation runs.

Agreed.

Reply:
No, that crashes for me. Does it work on your side?

Reply:
🤔

Reply:
Did some more experimentation, appears the `self.identifyScreenPoint = nil` prevents the crash.

Reply:
Me too confused by where the problem comes from. Maybe there is some magic with SwiftUI that caused the `PopupView` to be computed too early when the popup isn't non `nil` yet.

Reply:
@philium I'm curious if you see anything we might be missing here. Removing the following line in this branch will cause a crash upon selecting a feature.

https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/9d2e3b7d6080595e04da06777420852ee7714525/Shared/Samples/Display%20points%20using%20clustering%20feature%20reduction/DisplayPointsUsingClusteringFeatureReductionView.swift#L69

Reply:
Basically what is happening is that the floating panel's content is stale—an older version of the content closure is being evaluated. Why that is, I don't know. I do have a fix, but unfortunately it is in the toolkit, so you'll have to go with your workaround for now.

Reply:
Never mind. I thought I had a fix, but apparently not.

Reply:
For now, why not just conditionally unwrap the popup?

```swift
if let popup {
    // Create popup view.
}
```

It seems like that would be more straightforward than setting `identifyScreenPoint` to `nil`.

Reply:
@njarecha identified the issue during workshop this afternoon.

`identifyScreenPoint` which is of type `CGPoint` is being used to start the task. `CGPoint` doesn't conform to Equatable. 

The documentation of the task modifier's `id` parameter states:

> The value to observe for changes. The value must conform to the Equatable protocol.

Because of this, we need to manually reset `identifyScreenPoint` to avoid undefined behavior. Nimesh also suggested this reset should be `defer`red.

Reply:
That's quite surprising as well, as one of the [deprecated doc](https://developer.apple.com/documentation/coregraphics/1456179-cgpointequaltopoint) for CGPoint says

> The [CGPoint](https://developer.apple.com/documentation/corefoundation/cgpoint) type adopts the Equatable protocol

Anyway, great to learn the root cause and potentially something we can put into the onTapGesture's discussion. Thanks!

Reply:
> `CGPoint` doesn't conform to Equatable.

Though it doesn't show up in the doc as an explicit conformance, `CGPoint` most definitely does conform to `Equatable`, as seen in [this doc](https://developer.apple.com/documentation/coregraphics/1456179-cgpointequaltopoint):

![image](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/2257493/5f879184-acb7-4e90-98b8-ee96af8f0d18)

Not to mention that if it didn't, the compiler would throw an error because that constraint of the `id` parameter is enforced by the compiler.

> should be `defer`red.

A [`defer` statement](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Defer-Statement) ensures that whatever is inside the block is executed "just before transferring program control outside of the scope that the `defer` statement appears in," whichever way that is done. In this case, that happens as soon as the end of the `action` closure of the task modifier is reached. In other words, there's no difference between using a defer statement and moving the contents of the defer statement to end of the closure. And since the order of setting the state properties doesn't matter, using `defer` here is essentially pointless.

Reply:
Hm, this is interesting, so do you think this isn't truly the root cause? This line still makes the difference between crashing or not. Is there possibly a semantic difference between "adopts" and "conforms"? 

I often check the "Conforms To" section in the doc when looking for conformance so I'm a bit disappointed that it doesn't show here if it does conform to it.

<img width="370" alt="Screenshot 2023-08-17 at 14 24 50" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/701455a1-eba3-49c6-87c3-dcbb9ea1ac71">

Reply:
Note that I've also tried swapping out the Floating Panel for a sheet here and a sheet suffers the same crash if the point isn't reset.

Reply:
I'll do a little more investigating. I have another idea.

Reply:
I was able to write up a pretty compact reproducer.

<details>

```swift
import SwiftUI

struct ContentView: View {
    @State private var point: CGPoint?
    
    @State private var isPresented = false
    
    @State private var content: String?
    
    var body: some View {
        Text("Hello, world!")
            .onTapGesture {
                point = CGPoint(
                    x: Double(Int.random(in: 100...400)),
                    y: Double(Int.random(in: 500...900))
                )
            }
            .task(id: point) {
                guard point != nil else { return }
                point = nil                // <---- Will crash without this line commented
                content = "Some message"
                isPresented = true
            }
            .sheet(isPresented: $isPresented) {
                Text(content!)
            }
            .padding()
    }
}
```

</details>

Reply:
Found the explanation in this [thread](https://stackoverflow.com/questions/74554151/task-in-swiftui-runs-but-view-that-is-inside-a-sheet-is-not-updated).

If you change the `floatingPanel`'s closure  to capture the `popup` then it works as expected without setting `identifyScreenPoint = nil`.

```
                .floatingPanel(
                    selectedDetent: .constant(.half),
                    horizontalAlignment: .leading,
                    isPresented: $showsPopup
                ) { [popup] in
                    PopupView(popup: popup!, isPresented: $showsPopup)
                        .showCloseButton(true)
                        .padding()
                }
```

Reply:
It would be great if there was a version of the `floatingPanel` modifier that took an optional item as a data source, presenting the panel when the item is non-nil and providing it as a non-optional to the content closure, similar to [`sheet(item:onDismiss:content:)`](https://developer.apple.com/documentation/swiftui/view/sheet(item:ondismiss:content:)).

Reply:
> It would be great if there was a version of the floatingPanel modifier that took an optional item as a data source, presenting the panel when the item is non-nil and providing it as a non-optional to the content closure, similar to [sheet(item:onDismiss:content:)](https://developer.apple.com/documentation/swiftui/view/sheet(item:ondismiss:content:)).

I've been thinking about that this week. I'll create an issue for it.